src/lib/es5.d.ts

```diff
@@ -1728,12 +1724,12 @@ interface Int8Array {
     sort(compareFn?: (a: number, b: number) => number): this;
 
     /**
-      * Gets a new Int8Array view of the ArrayBuffer store for this array, referencing the elements
+      * Gets a new T view of the ArrayBuffer store for this array, referencing the elements
       * at begin, inclusive, up to end, exclusive.
       * @param begin The index of the beginning of the array.
       * @param end The index of the end of the array.
       */
-    subarray(begin: number, end?: number): Int8Array;
+    subarray(begin: number, end?: number): T;

```

Comment:
Rather than return `T`, either return `this` or return `TypedArray` with a subtyped overload in each specific typed-array constructor (e.g. return `Int32Array` in the `Int32Array` interface).